src/object/object_primitive.c

```diff
@@ -16374,3 +16381,225 @@ mr_get_compression_length (const char *string, int charlen)
+      length = (int) compression_length;
+      compressed = true;
+      str = compressed_string;
+    }
+  else
+    {
+      /* Compression failed */
+      length = str_length;
+      compression_length = 0;
+      str = string;
+    }
+
+  /* 
+   * Step 2 : Compute the disk size of the dbvalue.
+   * We are sure that the initial string length is greater than 255, which means that the new encoding applies.
+   */
+
+  switch (DB_VALUE_DOMAIN_TYPE (value))
+    {
+    case DB_TYPE_VARCHAR:
+    case DB_TYPE_VARNCHAR:
+      *val_size = or_packed_varchar_length (length + PRIM_TEMPORARY_DISK_SIZE) - PRIM_TEMPORARY_DISK_SIZE;
+      break;
+
+    default:
+      /* It should not happen */
+      assert (false);
+      rc = ER_FAILED;
+      goto cleanup;
+    }
+
+  /* Step 3 : Initialize the buffer */
+  OR_BUF_INIT (*buf, val_p, *val_size);
+
+  /* Step 4 : Insert the disk representation of the dbvalue in the buffer */
+
+  switch (DB_VALUE_DOMAIN_TYPE (value))
+    {
+    case DB_TYPE_VARNCHAR:
+    case DB_TYPE_STRING:
+      rc = mr_write_compressed_string_to_buffer (buf, str, (int) compression_length, str_length, align);
+      break;
+
+    default:
+      /* It should not happen. */
+      assert (false);
+      goto cleanup;
+    }
+
+cleanup:
+  buf->error_abort = error_abort;
+
+  if (compressed_string != NULL)
+    {
+      free_and_init (compressed_string);
+    }
+
+  if (wrkmem != NULL)
+    {
+      free_and_init (wrkmem);
+    }
+
+  if (rc == ER_TF_BUFFER_OVERFLOW)
+    {
+      return or_overflow (buf);
+    }
+
+  return rc;
+}
+
+/* mr_write_compressed_string_to_buffer()	  : Similar function to the previous implementation of 
+ *						    or_put_varchar_internal.
+ *
+ * buf(in/out)					  : Buffer to be written the string.
+ * compressed_string(in)			  : The string to be written.
+ * compressed_length(in)			  : Compressed length of the string. If it is 0, then no
+ *						    compression happened.
+ * decompressed_length(in)			  : Decompressed length of the string.
+ * align(in)					  :
+ */
+
+static int
+mr_write_compressed_string_to_buffer (OR_BUF * buf, char *compressed_string, int compressed_length,
+				      int decompressed_length, int align)
+{
+  int net_charlen = 0;
+  int rc = NO_ERROR;
+
+  assert (decompressed_length >= PRIM_MINIMUM_STRING_LENGTH_FOR_COMPRESSION);
+
+  /* store the size prefix */
+
+  rc = or_put_byte (buf, 0xFF);
+
+  if (rc != NO_ERROR)
+    {
+      return rc;
+    }
+  /* Store the compressed size */
+  OR_PUT_INT (&net_charlen, compressed_length);
+  rc = or_put_data (buf, (char *) &net_charlen, OR_INT_SIZE);
+  if (rc != NO_ERROR)
+    {
+      return rc;
+    }
+
+  /* Store the decompressed size */
+  OR_PUT_INT (&net_charlen, decompressed_length);
+  rc = or_put_data (buf, (char *) &net_charlen, OR_INT_SIZE);
+  if (rc != NO_ERROR)
+    {
+      return rc;
+    }
+
+  /* Get the string disk size */
+  if (compressed_length > 0)
+    {
+      net_charlen = compressed_length;
```

Comment:
@lucianlazar2211 

I'd like to propose to use another variable instead of `net_charlen` for readability.